tests/mongo-tools/dump_restore_test.go

```diff
@@ -21,57 +21,100 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-	"go.mongodb.org/mongo-driver/bson"
 
 	"github.com/FerretDB/dance/tests/common"
 )
 
 func TestDumpRestore(t *testing.T) {
 	ctx, db := common.Setup(t)
 
-	// TODO restart existing dump from https://github.com/mcampo2/mongodb-sample-databases instead
-	_, err := db.Collection("test").InsertOne(ctx, bson.D{{"foo", "bar"}})
+	dbName := "sample_geospatial"
+	db = db.Client().Database(dbName)
+
+	// cleanup database
+	err := db.Drop(ctx)
 	require.NoError(t, err)
 
-	localRoot := filepath.Join("..", "..", "dumps")
-	containerRoot := "/dumps/"
+	localActualPath := filepath.Join("..", "..", "dumps", "actual")
+	containerActualPath := filepath.Join("/dumps", "actual")
 
-	expectedState := getDatabaseState(t, ctx, db)
+	containerExpectedPath := filepath.Join("/dumps", "expected")
+	localExpectedPath := filepath.Join("..", "..", containerExpectedPath)
+
+	containerSourcePath := filepath.Join("/dumps", "mongodb-sample-databases", "dump")
 
 	// pre-create directories to avoid permission issues
-	err = os.Chmod(localRoot, 0o777)
+	err = os.Chmod(localActualPath, 0o777)
 	require.NoError(t, err)
-	err = os.RemoveAll(filepath.Join(localRoot, db.Name()))
+	err = os.Chmod(localExpectedPath, 0o777)
 	require.NoError(t, err)
-	err = os.Mkdir(filepath.Join(localRoot, db.Name()), 0o777) // 0o777 is typically downgraded to 0o755 by umask
+
+	err = os.RemoveAll(filepath.Join(localActualPath, dbName))
 	require.NoError(t, err)
-	err = os.Chmod(filepath.Join(localRoot, db.Name()), 0o777) // fix after umask
+	err = os.RemoveAll(filepath.Join(localExpectedPath, dbName))
 	require.NoError(t, err)
 
-	// dump a database
-	err = runDockerComposeCommand(
-		"mongodump",
-		"--out", containerRoot,
-		"--db", db.Name(),
-		"--verbose",
-		"mongodb://host.docker.internal:27017/",
-	)
+	err = os.Mkdir(filepath.Join(localActualPath, dbName), 0o777) // 0o777 is typically downgraded to 0o755 by umask
+	require.NoError(t, err)
+	err = os.Mkdir(filepath.Join(localExpectedPath, dbName), 0o777)
+	require.NoError(t, err)
+
+	err = os.Chmod(filepath.Join(localActualPath, dbName), 0o777) // fix after umask
 	require.NoError(t, err)
+	err = os.Chmod(filepath.Join(localExpectedPath, dbName), 0o777)
+	require.NoError(t, err)
+
+	// restore a database from a sample dump
+	mongorestore(t, dbName, containerSourcePath)
+
+	expectedState := getDatabaseState(t, ctx, db)
+
+	// "bootstrap" the expected dump from restored database
+	mongodump(t, dbName, containerExpectedPath)
 
 	// cleanup database
-	ctx, db = common.Setup(t)
+	require.NoError(t, db.Drop(ctx))
+
+	// restore a database from the expected dump
+	mongorestore(t, dbName, containerExpectedPath)
+
+	// dump a database
+	mongodump(t, dbName, containerActualPath)
+
+	// get database state after restore and compare it
+	actualState := getDatabaseState(t, ctx, db)
+	assert.Equal(t, expectedState, actualState)
```

Comment:
We don't need to do this comparison for this task, I think that was our discussion :hugs:

Reply:
I believe that discussion was focused more on testing if FerretDB omits any data from the external **source dump** generated by MongoDB. We've already merged the database state comparison with the past PRs, so I assumed it's still in a scope.

We aim to compare dumps created by FerretDB, but as binary files are... _just binary files_, the comparison of them is just a matter of saying "Yes they're equal" or "No, they're not". 

Comparing the database state under both of these dumps is just handful for the output purposes.

Let's maybe leave this conversation open to discuss it with everyone if needed :eyes:

Reply:
Yeah okie, let's keep it for now.